iot-e2e-tests/common/src/test/java/tests/integration/com/microsoft/azure/sdk/iot/iothub/setup/DirectMethodsCommon.java

```diff
@@ -169,18 +171,32 @@ public void dispose()
 
     private String loopback(Object methodData)
     {
-        String payload = new String((byte[])methodData, StandardCharsets.UTF_8).replace("\"", "");
+        String payload = new String(((DirectMethodPayload)methodData).getPayloadAsJsonString().getBytes(StandardCharsets.UTF_8)).replace("\"", "");
```

Comment:
Instead of casting methodData as a DirectMethodPayload object here, can you just change the parameter of this method to take a DirectMethodPayload instead? Here and below